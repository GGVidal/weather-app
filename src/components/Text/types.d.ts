import { ReactNode } from "react";
import { TextProps as RnTextProps } from "react-native";

export interface TextProps extends RnTextProps {
  fontSize: string;
  color: string;
  fontWeight: string;
  children: ReactNode;
  lineHeight?: string;
}
