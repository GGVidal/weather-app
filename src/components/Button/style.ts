import styled from "styled-components/native";
// import { ButtonProps } from "./types";

export const StyledButton = styled.Pressable`
  border-radius: 10px;
  background-color: rgba(255, 255, 255, 0.36);
  justify-content: center;
  align-items: center;
  padding-vertical: 12px;
`;
