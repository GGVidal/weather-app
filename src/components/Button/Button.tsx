import React, { FC } from "react";
import { PressableProps } from "react-native";
import { StyledButton } from "./style";

export const Button: FC<PressableProps> = ({ children, onPress }) => {
  return <StyledButton onPress={onPress}>{children}</StyledButton>;
};
