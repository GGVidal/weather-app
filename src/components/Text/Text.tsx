import React, { FC } from "react";
import { StyledText } from "./style";
import { TextProps } from "./types";

export const Text: FC<TextProps> = ({
  color,
  fontSize,
  fontWeight,
  letterSpacing,
  children,
  lineHeight,
}) => {
  return (
    <StyledText
      color={color}
      fontSize={fontSize}
      fontWeight={fontWeight}
      letterSpacing={letterSpacing}
      lineHeight={lineHeight}
    >
      {children}
    </StyledText>
  );
};
