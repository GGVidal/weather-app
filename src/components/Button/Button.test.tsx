import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import TestRenderer from "react-test-renderer";

import { Button } from "../Button/Button";
import { Text } from "../Text";

describe("renders correctly", () => {
  it("works", () => {
    const renderer = TestRenderer.create(
      <Button onPress={() => console.log("gg")} />
    );
    expect(renderer).toMatchSnapshot();
  });
});
