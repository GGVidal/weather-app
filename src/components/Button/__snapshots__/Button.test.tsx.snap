// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly works 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(255, 255, 255, 0.36)",
      "borderRadius": 10,
      "justifyContent": "center",
      "opacity": 1,
      "paddingVertical": 12,
    }
  }
/>
`;
